js/canvas.js

```diff
@@ -0,0 +1,367 @@
+    this.drawRectangles();
+    this.canvas.addEventListener('mousedown', this.mouseDownHandler.bind(this));
+    this.canvas.addEventListener('mousemove', this.mouseMoveHandler.bind(this));
+    this.canvas.addEventListener('mouseup', this.mouseUpHandler.bind(this));
+  }
+
+  setSize() {
+    const { innerWidth: width, innerHeight: height } = window;
+
+    this.canvas.width = width - canvasWindowIndent;
+    this.canvas.height = height - canvasWindowIndent;
+  }
+
+  setOffsets_X_and_Y() {
+    const domRect = canvas.getBoundingClientRect();
+    const { left, top } = domRect;
+
+    this.offsetX = left;
+    this.offsetY = top;
+  }
+
+  drawRectangles() {
+    this.clear();
+
+    arrayOfRectangles.forEach(rect => {
+      if (rect.isSticked && !rect.isCrossed) {
+        const { stickedX, stickedY, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(stickedX, stickedY, width, height);
+      } else {
+        const { x, y, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(x, y, width, height);
+      }
+    });
+  }
+
+  clear() {
+    const { width, height } = this.canvas;
+
+    this.ctx.clearRect(0, 0, width, height);
+  }
+
+  isRectangle_a_and_b_crossed(rectA, rectB) {
+    const topSideOfRectA = rectA.y;
+    const leftSideOfRectA = rectA.x;
+    const rightSideOfRectA = rectA.x + rectA.width;
+    const bottomSideOfRectA = rectA.y + rectA.height;
+    const topSideOfRectB = rectB.y;
+    const leftSideOfRectB = rectB.x;
+    const rightSideOfRectB = rectB.x + rectB.width;
+    const bottomSideOfRectB = rectB.y + rectB.height;
+
+    if (bottomSideOfRectA <= topSideOfRectB || rightSideOfRectA <= leftSideOfRectB) {
+      return false;
+    }
+
+    if (topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA) {
+      return false;
+    }
+
+    return true;
+  }
+
+  isRectangle_a_and_b_sticked(rectA, rectB) {
+    const topSideOfRectA = rectA.y - stickyDistance;
+    const leftSideOfRectA = rectA.x - stickyDistance;
+    const rightSideOfRectA = rectA.x + rectA.width + stickyDistance;
+    const bottomSideOfRectA = rectA.y + rectA.height + stickyDistance;
+    const topSideOfRectB = rectB.y - stickyDistance;
+    const leftSideOfRectB = rectB.x - stickyDistance;
+    const rightSideOfRectB = rectB.x + rectB.width + stickyDistance;
+    const bottomSideOfRectB = rectB.y + rectB.height + stickyDistance;
+
+    if (bottomSideOfRectA <= topSideOfRectB || rightSideOfRectA <= leftSideOfRectB) {
+      return false;
+    }
+
+    if (topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA) {
+      return false;
+    }
+
+    return true;
+  }
+
+  isCrossedRectangle(rectA) {
+    return arrayOfRectangles.reduce((acc, rectB) => {
+      if (rectA !== rectB && this.isRectangle_a_and_b_crossed(rectA, rectB)) acc = true;
+
+      return acc;
+    }, false);
+  }
+
+  isStickedRectangle(rectA) {
```

Comment:
`isCrossedRectangle` and `isStickedRectangle` looks like the same... maybe logic for  `isCrossed` can be combine with logic for `isSticked`